jaworski_cellseg/configuration_widget.py

```diff
@@ -0,0 +1,81 @@
+                data["inference_use_window_choice"] = jaworski_obj.inferer_widget.use_window_choice.isChecked()
+                data["inference_window_size"] = int(jaworski_obj.inferer_widget.window_size_choice.currentText())
+                data["inference_input"] = jaworski_obj.inferer_widget.model_input_size.value()
+                data["inference_overlap"] = jaworski_obj.inferer_widget.window_overlap_slider.value()
+                data["inference_perform_threhold"] = jaworski_obj.inferer_widget.thresholding_checkbox.isChecked()
+                data["inference_perform_threhold_value"] = jaworski_obj.inferer_widget.thresholding_slider.value()
+                data["inference_instance_segmentation"] = jaworski_obj.inferer_widget.use_instance_choice.isChecked()
+                data["inference_instance_segmentation_option"] = "Voronoi-Otsu"
+                voronoi_widget = jaworski_obj.inferer_widget.instance_widgets.methods[ "Voronoi-Otsu" ]
+                data["inference_instance_segmentation_spot_signma"] = voronoi_widget.counters[0].value()
+                data["inference_instance_segmentation_outline_signma"] = voronoi_widget.counters[1].value()
+                data["inference_instance_segmentation_small_object_removal"] = voronoi_widget.counters[2].value()
+                jaworski_obj.config['settings'][configuration_widget.save_confg.value] = data
+                yaml.dump(jaworski_obj.config, file)
+                ## Add configuration to the dropdown list
+                configuration_widget.configuration_names.choices = (
+                    list(configuration_widget.configuration_names.choices)
+                      + [configuration_widget.save_confg.value])
+    
+    def update_layer_dropdown(event, jaworski_obj):
+        print(f"Load configuration: {event} ")
+        configuration = jaworski_obj.config['settings'][event]
+        jaworski_obj.pre_process_data_widget.gaussian_checkbox.value = configuration["gausianFilter"]
+        jaworski_obj.pre_process_data_widget.gaussian_factor.value = configuration["gaussian_sigma"]
+        jaworski_obj.pre_process_data_widget.contrast_adjustment.value = configuration["contrast_adjustment"]
+        jaworski_obj.pre_process_data_widget.contrast_lower.value = configuration["contrast_min"]
+        jaworski_obj.pre_process_data_widget.contrast_upper.value = configuration["contrast_max"]
+        jaworski_obj.pre_process_data_widget.binary_map_threshold.value = configuration["binary_map_threshold"]
+        jaworski_obj.inferer_widget.use_window_choice.setChecked(configuration["inference_use_window_choice"])
+        jaworski_obj.inferer_widget.window_size_choice.setCurrentIndex(configuration["inference_window_size"])
+        jaworski_obj.inferer_widget.model_input_size.setValue(configuration["inference_input"])
+        jaworski_obj.inferer_widget.window_overlap_slider.setValue(configuration["inference_overlap"])
```

Comment:
I think you could define a setter and some mappings to make this a little less repetitive. Consider this for some inspiration.
```python
def set_widget_values(widget, config, mappings):
    for attr, key in mappings.items():
        setattr(widget, attr, config[key])

def update_layer_dropdown(event, jaworski_obj):
    print(f"Load configuration: {event}")
    configuration = jaworski_obj.config['settings'][event]
    
    pre_process_mappings = {
        'gaussian_checkbox.value': "gausianFilter",
        'gaussian_factor.value': "gaussian_sigma",
        'contrast_adjustment.value': "contrast_adjustment",
        'contrast_lower.value': "contrast_min",
        'contrast_upper.value': "contrast_max",
        'binary_map_threshold.value': "binary_map_threshold"
    }
    
    inferer_mappings = {
        'use_window_choice.setChecked': "inference_use_window_choice",
        'window_size_choice.setCurrentIndex': "inference_window_size",
        'model_input_size.setValue': "inference_input",
        'window_overlap_slider.setValue': "inference_overlap"
    }
    
    set_widget_values(jaworski_obj.pre_process_data_widget, configuration, pre_process_mappings)
    set_widget_values(jaworski_obj.inferer_widget, configuration, inferer_mappings)

Reply:
I applied your suggested changes.